cmd/security-secrets-setup/entrypoint.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/dumb-init /bin/sh
```

Comment:
it would be nice if instead of using dumb-init in the shebang we just use /bin/sh, and then call /usr/bin/dumb-init from the `ENTRPOINT` in the Dockerfile and use the `ENTRYPOINT` for the actual entrypoint.sh script as it currently is. This would allow us to try and use the same script in the snap, assuming that the environment variables are setup appropriately so we can override them from the snap.

However this is just a nice to have, since the snap already has a mostly working script, so I'm fine to push this off to a new issue if you just want to merge as-is to fix docker ASAP.

Reply:
This is actually modeled after the `docker-entrypoint.sh` that ships with Vault.  Since we're replacing the ENTRYPOINT to include a preprocessing script we are doing it the same way Vault is doing it.

Reply:
Right I understand that's how it's done in the vault Dockerfile, but I think my suggestion still stands as a possible improvement. I'm not sure of how to do this by inheriting the default args in the vault upstream Dockerfile, but you could certainly do something like:

```dockerfile
ENTRYPOINT ["/usr/bin/dumb-init"]
CMD ["/vault/our-entrypoint.sh", "server","-dev"]
```

(since the upstream dockerfile for vault at 1.0.1 has `CMD ["server","-dev"]` see https://github.com/hashicorp/docker-vault/blob/6ca0fa239f6a7150dfab827a0852a0031a02e028/0.X/Dockerfile#L80)

Reply:
Another way this could be done is to use a chain of scripts where at the end of the script, it executes all of it's arguments a la:

```bash
#!/bin/sh -e

# do stuff

exec "$@"
```

which you could then implement the desired effect with:

```dockerfile
ENTRYPOINT ["/usr/bin/dumb-init"]
CMD ["/vault/our-entrypoint.sh", "/usr/local/bin/upstream-vault-entrpoint.sh","server","-dev"]
```

Reply:
I think the way Vault's implementation looks cleaner to me since it centralizes the implementation in the shell script instead of spreading the logic into two places.

Reply:
so, Ian, why in the SNAP process, we do not use dumb-init to reap left-over zombie processes? If in the stand-alone, still having the same issue, I think we should use dumb-init the same way here.

Reply:
In snaps, services are implemented using systemd, which is a real init process and doesn't have the same issue

Reply:
ok, so the initial assumption is different then- in order to share the shell script code or at the most of codes, maybe can introduce another shell script and then from certain point of process, it calls into that shared script then. IMHO, however, since this is small enough script, I am not sure if it worth the efforts to do that kind of refactor or not (just for the sake of small duplication of codes?)

Reply:
As I said, it's probably not worth blocking this PR on refactoring to handle the snap as well, but the benefit is that both docker and the snap are calling `security-secrets-setup` in the same manner, which maintains consistency between the two. If we find a bug or problem with how security-secrets-setup is being called, with a shared script like this we reduce the amount of work it takes to fix and update that and greatly reduce the risk that one distribution method works and the other one doesn't. But since the next rewrite of this code will likely be to just call security-secrets-setup directly instead, I think it's fine to have 2 different implementations for now.

Reply:
The goal that I am trying to steer us towards is that we run security-secrets-setup in its own container after merging it with pki-init.  We would then run a stock version of Vault and override its RUN command to block on PKI availability instead of making the custom version that we do now.

Reply:
Sure, I understand this PR is just meant to be temporary, if that's your plan and it's too much work then feel free to ignore this suggestion for the time being.